### PR TITLE
Add progress bar for `mcap convert`

### PR DESCRIPTION
**Pull request by narasaka**
### Public-Facing Changes

This adds a progress bar when running `mcap convert ...`

### CODE REVIEW

File: go/ros/bag2mcap.go [hunk 1]
```diff
@@ -225,7 +232,7 @@ func channelIDForConnection(connID uint32) (uint16, error) {
 	return uint16(connID), nil
 }
 
-func Bag2MCAP(w io.Writer, r io.Reader, opts *mcap.WriterOptions) error {
+func Bag2MCAP(w io.Writer, r io.Reader, opts *mcap.WriterOptions, progressBar *progressbar.ProgressBar) error {
```
Comment (narasaka) [inline go/ros/bag2mcap.go:235]: I feel like i should just put the progress bar in this function instead of injecting it 🤷

**Comment by wkalt** [comment]
This looks great generally. The thing that's not clear to me is this:

```
 100% |████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| (1606/1606, 4395 it/s)

```

the 1606 needs some units, and I have no idea what "it/s" is. I guess that's the conversion rate? We can probably either remove that, or change it to a MB/sec computation.

File: go/ros/bag2mcap.go (hunk 1 above)
Comment (wkalt) [reply go/ros/bag2mcap.go:235]: that does seem nicer, but you'd just need to pass in a messageCount.

Maybe the cleanest option would be to pass in the `*Info` struct, and then construct the progress bar in here? You need to call Info() outside this function (as you do) or else you would need to pass this a ReadSeeker, which you're not in position to do.

File: go/ros/ros2db3_to_mcap.go [hunk 2]
```diff
@@ -263,6 +264,12 @@ func DB3ToMCAP(w io.Writer, db *sql.DB, opts *mcap.WriterOptions, searchdirs []s
 	if err != nil {
 		return err
 	}
+
+	progressBar := progressbar.NewOptions64(
+		int64(len(topics)),
```
Comment (wkalt) [inline go/ros/ros2db3_to_mcap.go:269]: what is len(topics) doing here?

Comment (narasaka) [reply go/ros/ros2db3_to_mcap.go:269]: isn't number of topics the denominator of the progress?

File: go/ros/bag2mcap.go [hunk 3]
```diff
@@ -173,6 +173,10 @@ func processBag(
 			}
 		}
 
+		if err != nil {
+			return err
+		}
```
Comment (wkalt) [inline go/ros/bag2mcap.go:178]: remove

File: go/ros/ros2db3_to_mcap.go [hunk 4]
```diff
@@ -313,6 +319,10 @@ func DB3ToMCAP(w io.Writer, db *sql.DB, opts *mcap.WriterOptions, searchdirs []s
 		if err != nil {
 			return fmt.Errorf("failed to write channel info: %w", err)
 		}
+
+		if err != nil {
+			return fmt.Errorf("failed to increment progress bar: %w", err)
+		}
```
Comment (wkalt) [inline go/ros/ros2db3_to_mcap.go:325]: remove

File: go/go.work [hunk 5]
```diff
@@ -1,4 +1,6 @@
-go 1.18
+go 1.21
```
Comment (wkalt) [inline go/go.work:1]: 1.18

**Review by wkalt — APPROVED** [review]

**Pull request merged**